src/main/java/sirius/biz/storage/layer3/VirtualFile.java

```diff
@@ -0,0 +1,794 @@
+     */
+    public VirtualFile parent() {
+        return parent;
+    }
+
+    /**
+     * Returns the absolute path of this file within the virtual file system.
+     *
+     * @return the absolute path of this file
+     */
+    public String path() {
+        if (parent != null) {
+            if (parent.name == null) {
+                return "/" + name;
+            } else {
+                return parent.path() + "/" + name;
+            }
+        }
+
+        return name();
+    }
+
+    /**
+     * Returns the name of this file.
+     *
+     * @return the name of this file
+     */
+    public String name() {
+        return name == null ? "/" : name;
+    }
+
+    /**
+     * Returns a short description of the file.
+     *
+     * @return the description of the file
+     */
+    @Nullable
+    public String description() {
+        return NLS.smartGet(description);
+    }
+
+    /**
+     * Returns the last modification timestamp.
+     *
+     * @return the last modification as in {@link File#lastModified()}
+     */
+    public long lastModified() {
+        try {
+            return lastModifiedSupplier == null ? 0 : lastModifiedSupplier.apply(this);
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "lastModifiedSupplier");
+        }
+    }
+
+    /**
+     * Returns the last modification timestamp as {@link LocalDateTime}.
+     *
+     * @return the last modification timestamp
+     */
+    public LocalDateTime lastModifiedDate() {
+        try {
+            return lastModifiedSupplier == null ?
+                   null :
+                   Instant.ofEpochMilli(lastModifiedSupplier.apply(this))
+                          .atZone(ZoneId.systemDefault())
+                          .toLocalDateTime();
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "lastModifiedSupplier");
+        }
+    }
+
+    /**
+     * Returns the size of this file.
+     *
+     * @return the size in bytes
+     */
+    public long size() {
+        try {
+            return sizeSupplier == null ? 0 : sizeSupplier.apply(this);
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "sizeSupplier");
+        }
+    }
+
+    /**
+     * Determines if this file can (probably) be deleted.
+     *
+     * @return <tt>true</tt> if this file can be delete or <tt>false</tt> otherwise
+     */
+    public boolean canDelete() {
+        try {
+            if (canDeleteHandler != null) {
+                return canDeleteHandler.test(this);
+            }
+
+            return deleteHandler != null;
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "canDeleteHandler");
+        }
+    }
+
+    /**
+     * Tries to delete this file.
+     *
+     * @return <tt>true</tt> if the operation was successful, <tt>false</tt> otherwise
+     */
+    public boolean tryDelete() {
+        try {
+            if (deleteHandler == null) {
```

Comment:
I think inverting this if check would improve readability/flow and woudl be consistent to the other methods in this class